includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -759,7 +759,7 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 				sprintf( __( 'Payment failed: %s', 'woocommerce-gateway-stripe' ), $e->getLocalizedMessage() )
 			);
 
-			return [ 'result' => 'failure' ];
```

Comment:
I changed this to be inline with my understanding of how the `process_payment()` function is intended to run. 

If you don't throw the exception here you get the following PHP error using the block checkout when an payment error occurs, and there's no detailed error message shown on the checkout. You just get a generic "something went wrong".  

```
PHP Warning:  Undefined array key "redirect" in app/public/wp-content/plugins/woocommerce/src/StoreApi/Legacy.php on line 68
```

Just noting that this change is causing the PHP tests to fail. I'll look into fixing those tomorrow.

Reply:
I ended up reversing this change and going with a different approach to ensuring error messages are displayed on the block checkout. see https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2840/commits/a68977db6eef7afd26a947ed804b3233a8191db6.